src/lib.rs

```diff
@@ -150,6 +160,23 @@ impl SecretClient {
         Ok(secrets)
     }
 
+    #[cfg(feature = "read")]
+    async fn read_secrets_internal_option<A>(&self, path: &str) -> Result<Option<A>>
+    where
+        A: for<'de> Deserialize<'de>,
+    {
+        use vaultrs::error::ClientError;
+
+        let secrets: std::result::Result<A, vaultrs::error::ClientError> =
+            kv2::read::<A>(&self.client, &self.mount, path).await;
+
+        match secrets {
+            Ok(data) => Ok(Some(data)),
+            Err(ClientError::APIError { code: 404, errors: _ }) => Ok(None),
```

Comment:
This is somewhat strange to me and I think a client error should be handled on the caller side. `Result<Option, _>` is a not optimal return type (like `Either[_, Option]`) and it hides too much for me in terms of the functionality of a crate.

Reply:
Yeah, I think having an Option wrapped in a Result complicates the API. One has to pattern match on both type to get to the actual value, which is not ergonomic. I would suggest to not introduce an Option and keep the Result type as it is currently.

Reply:
Not finding something is for me not necessarily a client issue but I get your point that this is coming from a HTTP client error. 
It reflects a valid state of the system though and I see this more as an `IO[Option[T]]`. 
I guess the underlying question is if we find it useful to offer this as part of the `vaultier` interface or just see this as "advanced" and want the users to deal with `vaultrs` implementation details again in those cases?
A middle ground could be to introduce something like `VaultierError::NotFound`?

Reply:
I mean, it is a 404 in the range of 4xx, so client errors. When you ask for something in a path that does not exist, it is "your" fault, would be my opinion and is the same behavior in http APIs. And then it could be just the secret or the whole path that is not found.

Reply:
> A middle ground could be to introduce something like `VaultierError::NotFound`?

Not having looked into it a lot, this sounds reasonable, it also addresses the concerns w.r.t. ergonomics, I'd say.